### ISSUE TITLE
Support tuple as dims parameter for grid

### ISSUE THREAD

**Issue by karlwessel** [JuliaGraphs/LightGraphs.jl#1260]
If i want to create a grid graph representing connections between elements in an existing array i have to write

```
julia> using LightGraphs

julia> data = rand(5,3)
5×3 Array{Float64,2}:
 0.110087  0.960253   0.00778524
 0.232397  0.645722   0.688389  
 0.850882  0.809089   0.949893  
 0.46178   0.832621   0.0349758 
 0.742825  0.0598922  0.928002  

julia> grid(collect(size(data)))
{15, 22} undirected simple Int64 graph
```
it would be a bit nicer and more readable if I could just write `grid(size(data))` instead of `grid(collect(size(data)))`.

This is a really minor improvement and I am willing to create a PR myself for it as soon as I have a bit of time and nobody has objections. Just wanted to record it here.

Are there any other methods like `grid` that would benefit from this change?

**Comment by sbromberger** [comment]
No objections to you making a PR for this. I think it makes sense.

**Comment by sbromberger** [comment]
Closed via #1262
